haas/api.py

```diff
@@ -505,21 +563,72 @@ def headnode_detach_network(headnode, hnic):
                             # Network Code #
                             ################
 
+@rest_call('GET', '/networks')
+def list_networks():
+    """Lists all networks
+
+    """
+    get_auth_backend().require_admin()
+
+    networks = db.session.query(model.Network).all()
+    result = {}
+    for n in networks:
+        if n.access:
+            net = {'network_id': n.network_id, 'projects': [p.label for p in n.access]}
+        else:
+            net = {'network_id': n.network_id, 'projects': None}
+        result[n.label] = net
+            
+    return json.dumps(result, sort_keys = True)
+
+@rest_call('GET', '/network/<network>/attachments')
+def list_network_attachments(network, project=None):
+    """Lists all the attachments from <project> for <network>
+
+    If <project> is not specified, lists all attachments for <network>
+    """
+    auth_backend = get_auth_backend()
+    network = _must_find(model.Network, network)
+    
+    attachments = network.attachments
+    nodes = {}
+    authorized = auth_backend.have_project_access(network.owner)
+
+    if project is not None:
+        project = _must_find(model.Project, project)
+        for proj in network.access:
```

Comment:
Instead of a for loop here, is there a reason not to use:
```python
if not (project in network.access and auth_backend.have_project_access(project)):
    raise ...
```